pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/phimanifest/PhiManifestService.java

```diff
@@ -0,0 +1,267 @@
+                ddpInstanceDto.getEsParticipantIndex(), ddpParticipantId);
+        return participant != null && hasParticipantConsentedToSharedLearning(participant, ddpInstanceDto)
+                && hasParticipantConsentedToTumor(participant);
+    }
+
+    private boolean hasParticipantConsentedToTumor(ElasticSearchParticipantDto participant) {
+        return participant.getDsm().isPresent() && participant.getDsm().get().isHasConsentedToTissueSample();
+    }
+
+    private boolean hasAdultParticipantConsentedToTumor(ElasticSearchParticipantDto participant, String studyGuid) {
+        if (DBConstants.LMS_STUDY_GUID.equals(studyGuid)) {
+            return checkAnswerToActivity(participant, CONSENT_ADDENDUM_ACTIVITY_STABLE_ID, LMS_QUESTION_SOMATIC_CONSENT_ADDENDUM_TUMOR,
+                    true);
+        } else if (DBConstants.OSTEO_STUDY_GUID.equals(studyGuid)) {
+            return checkAnswerToActivity(participant, CONSENT_ADDENDUM_ACTIVITY_STABLE_ID, OS2_QUESTION_SOMATIC_CONSENT_ADDENDUM_TUMOR,
+                    true);
+        }
+        return false;
+    }
+
+    private boolean hasParticipantConsentedToSharedLearning(ElasticSearchParticipantDto participant, DDPInstanceDto ddpInstanceDto) {
+        if (participant.getDsm().isEmpty()) {
+            log.error("Participant {} does not have a DoB, which means we can't assess their consent", participant.getParticipantId());
+            return false;
+        }
+        String dateOfBirth = participant.getDsm().get().getDateOfBirth();
+        String dateOfMajority = (String) participant.getDsm().get().getDateOfMajority();
+        if (StringUtils.isBlank(dateOfMajority) || DateTimeUtil.isAdult(dateOfMajority)) {
+            return hasAdultParticipantConsentedToTumor(participant, ddpInstanceDto.getStudyGuid());
+        }
+        int age = DateTimeUtil.calculateAgeInYears(dateOfBirth);
+        if (age >= 7) {
+            return checkAnswerToActivity(participant, CONSENT_ADDENDUM_PEDIATRICS_ACTIVITY_STABLE_ID,
+                    SOMATIC_CONSENT_TUMOR_PEDIATRIC_QUESTION, true) && checkAnswerToActivity(participant,
+                    CONSENT_ADDENDUM_PEDIATRICS_ACTIVITY_STABLE_ID, SOMATIC_ASSENT_ADDENDUM_QUESTION, true);
+
+        } else if (age < 7) {
+            return checkAnswerToActivity(participant, CONSENT_ADDENDUM_PEDIATRICS_ACTIVITY_STABLE_ID,
+                    SOMATIC_CONSENT_TUMOR_PEDIATRIC_QUESTION, true);
+        }
+        return false;
+    }
+
+    /**
+     * checks if the answer provided by a participant to a survey question matches the criteria
+     *
+     * @param participant  participant data from ES.
+     * @param activityName stable id of the activity
+     * @param question     stable id of the question
+     * @param answer       desired answer
+     */
+    private boolean checkAnswerToActivity(ElasticSearchParticipantDto participant, String activityName, String question, Object answer) {
```

Comment:
Can you move this to `ElasticSearchParticipantDto` and rename to something like `hasAnswerInActivity()`?